src/gordon_gcp/clients/auth.py

```diff
@@ -190,17 +190,30 @@ async def refresh_token(self):
     async def _refresh_token_using_compute_credentials(self):
         metadata_host = os.getenv(environment_vars.GCE_METADATA_ROOT,
                                   'metadata.google.internal')
-        url = (f'http://{metadata_host}/computeMetadata/'
-               'v1/instance/service-accounts/'
-               f'{self.creds._service_account_email}/token')
-        headers = {'metadata-flavor', 'google'}
-        await self._execute_refresh_token_request(url, 'GET', headers)
+
+        headers = {'Metadata-Flavor': 'Google'}
+        sa_url = (f'http://{metadata_host}/computeMetadata/'
+                  'v1/instance/service-accounts/'
+                  f'{self.creds._service_account_email}/'
+                  '?recursive=true')
+
+        sa_response = await self._execute_request(sa_url, 'GET', headers)
+        email = (sa_response['email']
+                 if 'email' in sa_response
+                 else self.creds._service_account_email)
```

Comment:
this might be clearer as a standard if-block. i think we lose any of the conciseness we might have gotten from a ternary operator when it ends up extending to same number of lines.

Reply:
Yeah, the else just reasign the same as before. I think depending on the state this can work sometimes, but I mostly just took that approach by looking at the refresh function of the compute engine credentials object, where they do the same thing.